compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -3390,4 +3390,33 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             err.span_label(block.span, "this block is missing a tail expression");
         }
     }
+
+    pub(crate) fn suggest_swapping_lhs_and_rhs(
+        &self,
+        err: &mut Diag<'_>,
+        checked_ty: Ty<'tcx>,
+        expected_ty_expr: &'tcx hir::Expr<'tcx>,
+        op: hir::BinOp,
+    ) {
+        match op.node {
+            hir::BinOpKind::Eq => {
+                if let Some(partial_eq_def_id) = self.infcx.tcx.lang_items().eq_trait()
+                    && let lhs_ty = self.check_expr(expected_ty_expr)
```

Comment:
we already know `lhs_ty` within `check_overloaded_binop` so there's no need to recheck the expression here again.